### PR TITLE
Fix unnecessary newline addition on sub-tables

### LINKED ISSUE #69: Every run of pre-commit adds a newline after [tool.mypy]
Hi.

I think that https://github.com/tox-dev/pyproject-fmt/pull/67 added a bug where it keep adds a newline.

After several runs, I get the following:
```toml
[[tool.mypy.overrides]]
module = "vendor.*"
ignore_errors = true

















[tool.ruff]
exclude = [
  ".git",
````

Message after a run:
```
$ pyproject-fmt pyproject.toml
--- pyproject.toml

+++ pyproject.toml

@@ -174,6 +174,7 @@

 
 
 
+
 [tool.ruff]
 exclude = [
   ".git",
```

I'll gladly provide more if you need it.

### PR DESCRIPTION
**Pull request by adamchainz**
Fixes #69.

### CODE REVIEW

**Comment by gaborbernat** [comment]
Can you add a test that for this new case if there's no newline we add it?

**Comment by adamchainz** [comment]
Try that!

File: tests/formatter/test_tools.py [hunk 1]
```diff
@@ -48,3 +48,40 @@ def test_tools_ordering(fmt: Fmt) -> None:
     a = 0
         """
     fmt(fmt_tools, content, expected)
+
+
+def test_sub_table_newline(fmt: Fmt) -> None:
+    content = """
+    [tool.mypy]
+    a = 0
+
```
Comment (gaborbernat) [inline tests/formatter/test_tools.py:57]: Can we remove this newline too here 😅

Comment (gaborbernat) [reply tests/formatter/test_tools.py:57]: Oh, I see why you did not do it. 🤔 If you remove this line this patch will not add the newline in.

**Comment by adamchainz** [comment]
I've pushed an ignore for the coverage issue ( https://github.com/nedbat/coveragepy/issues/1480 ). We could make pyproject-fmt format all space between tables and sub-tables, but perahps first release this to fix the bug?

**Comment by MindTooth** [comment]
thank you

**Pull request merged**